ksql-engine/src/main/java/io/confluent/ksql/physical/PhysicalPlanBuilder.java

```diff
@@ -360,5 +359,11 @@ private KafkaStreams buildStreams(
     );
     return kafkaStreamsBuilder.buildKafkaStreams(builder, new StreamsConfig(newStreamsProperties));
   }
+
+  // protected because of test
+  protected String getServiceId() {
```

Comment:
You could drop the `protected`, thus making this package private. Then the test (which is in the same package) can access it.